mesonbuild/modules/python.py

```diff
@@ -334,6 +335,28 @@ def __init__(self, name: str, command: T.Optional[T.List[str]] = None,
             'version': '0.0',
         }
 
+    def sanity(self) -> 'PythonExternalProgram':
+        # Sanity check, we expect to have something that at least quacks in tune
+        cmd = self.get_command() + ['-c', INTROSPECT_COMMAND]
+        p, stdout, stderr = mesonlib.Popen_safe(cmd)
+        try:
+            info = json.loads(stdout)
+        except json.JSONDecodeError:
+            info = None
+            mlog.debug('Could not introspect Python (%s): exit code %d' % (str(p.args), p.returncode))
+            mlog.debug('Program stdout:\n')
+            mlog.debug(stdout)
+            mlog.debug('Program stderr:\n')
+            mlog.debug(stderr)
+
+        if isinstance(info, dict) and 'version' in info:
```

Comment:
I really feel like we should be checking separately whether the info is None or whether it's a dictionary. `None` is completely expected, but if we got a list her that would be a programming error that we'd be silently hiding.

Reply:
The original check (which I simply refactored), didn't try to "check for programming errors", only expected errors, i.e. it has to either be None or a json.loads which should always be a dict.

Reply:
`if info is not None` -> access it while assuming it is a dict. Programming errors will result in tracebacks now.